--- a/src/compiler/glsl/tests/cache_test.c	2020-03-18 22:24:18.000000000 +0100
+++ b/src/compiler/glsl/tests/cache_test.c	2020-03-30 12:09:05.000000000 +0200
@@ -132,8 +132,8 @@
 {
    bool sub_dirs_created = false;
 
-   char buf[PATH_MAX];
-   if (getcwd(buf, PATH_MAX)) {
+   char *buf = getcwd(NULL, 0);
+   if (buf) {
       char *full_path = NULL;
       if (asprintf(&full_path, "%s%s", buf, ++cache_dir) != -1 ) {
          struct stat sb;
@@ -142,6 +142,7 @@
 
          free(full_path);
       }
+      free(buf);
    }
 
    expect_true(sub_dirs_created, "create sub dirs");
--- a/src/util/process_test.c
+++ b/src/util/process_test.c
@@ -36,6 +36,10 @@
 #define PATH_MAX MAX_PATH
 #endif
 
+#if !defined(PATH_MAX) && defined(__GNU__)
+#define PATH_MAX (4096)
+#endif
+
 bool error = false;
 
 static void
