--- evdev.lst	2019-02-11 20:25:45.000000000 +0100
+++ evdev.lst.chromebook	2019-10-09 17:24:33.479737679 +0200
@@ -187,7 +187,15 @@
   teck227         Truly Ergonomic 227
   teck229         Truly Ergonomic 229
   apex300         SteelSeries Apex 300 (Apex RAW)
-  chromebook      Chromebook
+  chromebook      Chromebook (most models) | Search overlay
+  chromebook_ralt Chromebook (most models) | Right alt overlay
+  chromebook_m    Chromebook (most models) | Search overlay | F keys mapped to media keys
+  chromebook_m_ralt Chromebook (most models) | Right alt overlay | F keys mapped to media keys
+  chromebook_falco Chromebook Falco/Pixel/Pixel2 | Search overlay
+  chromebook_falco_ralt Chromebook Falco/Pixel/Pixel2 | Right alt overlay
+  chromebook_m_falco Chromebook Falco/Pixel/Pixel2 | Search overlay | F keys mapped to media keys
+  chromebook_m_falco_ralt Chromebook Falco/Pixel/Pixel2 | Right alt overlay | F keys mapped to media keys
+  chromebook_no_m Chromebook (most models) | No overlay | F keys mapped to media keys
   teck227         Truly Ergonomic Computer Keyboard Model 227 (Wide Alt keys)
   teck229         Truly Ergonomic Computer Keyboard Model 229 (Standard sized Alt keys, additional Super and Menu key)
 
