diff --git a/block/partitions/efi.c b/block/partitions/efi.c
index dc51f46..4bf0f97 100644
--- a/block/partitions/efi.c
+++ b/block/partitions/efi.c
@@ -112,7 +112,11 @@
 static int __init
 force_gpt_fn(char *str)
 {
-	force_gpt = 1;
+	/* Do not force GPT even if 'gpt' command line option is
+	 * specified for ChromeOS kernel.
+	 */
+	force_gpt = 0;
+	pr_warn("Not forcing GPT even though 'gpt' specified on cmd line.\n");
 	return 1;
 }
 __setup("gpt", force_gpt_fn);
@@ -623,7 +627,7 @@
 		good_agpt = is_gpt_valid(state,
 					 le64_to_cpu(pgpt->alternate_lba),
 					 &agpt, &aptes);
-        if (!good_agpt && force_gpt)
+	if (!good_agpt && force_gpt)
                 good_agpt = is_gpt_valid(state, lastlba, &agpt, &aptes);
 
         /* The obviously unsuccessful case */

