diff --git a/block/partitions/efi.c b/block/partitions/efi.c
index b33671d..068d5a6 100644
--- a/block/partitions/efi.c
+++ b/block/partitions/efi.c
@@ -369,7 +369,9 @@
 
 	/* Check the GUID Partition Table signature */
 	if (le64_to_cpu((*gpt)->signature) == GPT_HEADER_SIGNATURE_IGNORED) {
-		pr_debug("GUID Partition Table at LBA %llu marked IGNOREME\n");
+		pr_debug("GUID Partition Table at LBA %llu marked IGNOREME\n",
+			 (unsigned long long)lba);
+
 		if (ignored)
 			*ignored = 1;
 		goto fail;

