diff --git a/usr/bin/xflock4 b/usr/bin/xflock4
index d55cbf7..b7b28c6
--- a/usr/bin/xflock4
+++ b/usr/bin/xflock4
@@ -26,6 +26,7 @@ export PATH
 
 # Lock by xscreensaver or gnome-screensaver, if a respective daemon is running
 for lock_cmd in \
+    "env XSECURELOCK_PASSWORD_PROMPT=asterisks xsecurelock"\
     "xscreensaver-command -lock" \
     "light-locker-command --lock" \
     "gnome-screensaver-command --lock" \

